toolchain/check/class.h

```diff
@@ -0,0 +1,32 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_CHECK_CLASS_H_
+#define CARBON_TOOLCHAIN_CHECK_CLASS_H_
+
+#include "toolchain/check/context.h"
+
+namespace Carbon::Check {
+
+// If `type_id` is a class type, get its corresponding `SemIR::Class` object.
+// Otherwise returns `nullptr`.
+auto TryGetAsClass(Context& context, SemIR::TypeId type_id) -> SemIR::Class*;
+
+// Builds the `Self` type using the resulting type constant.
+auto SetNewClassSelfTypeId(Context& context, SemIR::ClassId class_id) -> void;
```

Comment:
I think this is actually just getting the `Self` type and assigning it, since it could be implemented using `GetClassType`. Maybe "Sets" is the right action here?

Also, what is "the resulting type constant" referring to? Maybe:

```suggestion
// Sets the `Self` type for the class.
auto SetClassSelfType(Context& context, SemIR::ClassId class_id) -> void;
```

Reply:
Done.